### PR TITLE
[OJS][main] #7337 Remove ReviewerSubmission/ReviewerSubmissionDAO

### PR DESCRIPTION
**Pull request by defstat**

### CODE REVIEW

File: pages/reviewer/ReviewerHandler.php [hunk 1]
```diff
@@ -88,9 +90,14 @@ protected function _validateAccessKey($request)
...
-        $reviewerSubmissionDao = DAORegistry::getDAO('ReviewerSubmissionDAO'); /** @var ReviewerSubmissionDAO $reviewerSubmissionDao */
-        $reviewerSubmission = $reviewerSubmissionDao->getReviewerSubmission($reviewId);
-        if (!$reviewerSubmission) {
+        $reviewAssignmentDao = DAORegistry::getDAO('ReviewAssignmentDAO'); /** @var ReviewAssignmentDAO $reviewAssignmentDao */
+        $reviewAssignment = $reviewAssignmentDao->getById($reviewId);
+        if (!$reviewAssignment) {
+            return;
+        } // e.g. deleted review assignment
+
+        $reviewSubmission = Repo::submission()->get($reviewAssignment->getSubmissionId());
```
Comment (NateWr) [inline pages/reviewer/ReviewerHandler.php:99]: Get the submission from the authorized context objects to make sure the user is authorized to access it:

```php
$reviewSubmission = $this->getAuthorizedContextObject(ASSOC_TYPE_SUBMISSION);
```

For extra protection, you can check that the submission ID matches the submission id of the review assignment.

File: pages/reviewer/ReviewerHandler.php [hunk 2]
```diff
@@ -116,7 +123,8 @@ protected function _validateAccessKey($request)
         // Register the user object in the session
         $reason = null;
         if (Validation::registerUserSession($user, $reason)) {
-            $this->submission = $reviewerSubmission;
+            // TODO:: WHAT ARE THOSE?
```
Comment (NateWr) [inline pages/reviewer/ReviewerHandler.php:126]: Can you clarify your question?

**Review by NateWr — CHANGES_REQUESTED** [review]
Looks good! Just a couple comments.

File: pages/reviewer/ReviewerHandler.php (hunk 2 above)
Comment (defstat) [reply pages/reviewer/ReviewerHandler.php:126]: It was my note in order to check what the `Validation::registerUserSession($user, $reason)` and then check what it means for the `$this->submission` not to be initialised.

Removed it

File: pages/reviewer/ReviewerHandler.php (hunk 1 above)
Comment (defstat) [reply pages/reviewer/ReviewerHandler.php:99]: Changed it. I added another check like so 

```
$reviewSubmission = $this->getAuthorizedContextObject(Application::ASSOC_TYPE_SUBMISSION);
$reviewAssignmentSubmission = Repo::submission()->get($reviewAssignment->getSubmissionId());
if (!$reviewSubmission || ($reviewSubmission->getId() != $reviewAssignmentSubmission->getId())) {
   return;
} // e.g. deleted review assignment
```

Comment (NateWr) [reply pages/reviewer/ReviewerHandler.php]: You don't need the second line. You should be able to do just:

```php
$reviewSubmission = $this->getAuthorizedContextObject(Application::ASSOC_TYPE_SUBMISSION);
if (!$reviewSubmission || $reviewSubmission->getId() != $reviewAssignment->getSubmissionId()) {
   return;
}
```

Comment (defstat) [reply pages/reviewer/ReviewerHandler.php]: shouldn't we make sure that `$reviewAssignment` submission exists?

Comment (NateWr) [reply pages/reviewer/ReviewerHandler.php]: If the ID is identical then `$this->getAuthorizedContextObject(Application::ASSOC_TYPE_SUBMISSION);` already checks that it exists.

Comment (defstat) [reply pages/reviewer/ReviewerHandler.php]: ok fixed

**Comment by NateWr** [comment]
Cherry-picked to `main` due to merge conflict.

**Pull request closed without merging**